File: src/jk_5/nailed/map/gameloop/instructions/InstructionSetDifficulty.java
package jk_5.nailed.map.gameloop.instructions;

import jk_5.nailed.map.gameloop.GameThread;
import jk_5.nailed.map.gameloop.IInstruction;
import net.minecraft.src.WorldServer;

/**
 * No description given
 *
 * @author jk-5
 */
public class InstructionSetDifficulty implements IInstruction {

    private int difficulty;

    @Override
    public void injectArguments(String arguments) {
        this.difficulty = Integer.parseInt(arguments);
    }

    @Override
    public void execute(GameThread controller) throws InterruptedException {
        WorldServer server = controller.getMap().getWorld();
        server.difficultySetting = this.difficulty;
        server.setAllowedSpawnTypes(controller.getMap().getMappack().shouldSpawnHostileMobs() && this.difficulty > 0, controller.getMap().getMappack().shouldSpawnFriendlyMobs());
    }
}

File: src/main/scala/jk_5/nailed/map/gameloop/instructions/InstructionSetDifficulty.scala
package jk_5.nailed.map.gameloop.instructions

import jk_5.nailed.map.gameloop.GameThread
import jk_5.nailed.map.gameloop.IInstruction
import net.minecraft.src.WorldServer

/**
 * No description given
 *
 * @author jk-5
 */
class InstructionSetDifficulty extends IInstruction {
  private var difficulty: Int = 0
  def injectArguments(arguments: String) = this.difficulty = Integer.parseInt(arguments)
  def execute(controller: GameThread) {
    val server: WorldServer = controller.getMap.getWorld
    server.difficultySetting = this.difficulty
    server.setAllowedSpawnTypes(controller.getMap.getMappack.shouldSpawnHostileMobs && this.difficulty > 0, controller.getMap.getMappack.shouldSpawnFriendlyMobs)
  }
}
